ibm/service/kubernetes/resource_ibm_container_vpc_worker_pool.go

```diff
@@ -453,28 +453,47 @@ func resourceIBMContainerVpcWorkerPoolUpdate(d *schema.ResourceData, meta interf
 			}
 		}
 	}
+
 	return resourceIBMContainerVpcWorkerPoolRead(d, meta)
 }
 
-func expandWorkerPoolTaints(d *schema.ResourceData, meta interface{}, clusterNameOrID, workerPoolName string) v2.WorkerPoolTaintRequest {
+func updateWorkerpoolTaints(d *schema.ResourceData, meta interface{}, clusterNameOrID string, workerPoolName string, taints []interface{}) error {
+
+	taintParam := expandWorkerPoolTaints(clusterNameOrID, workerPoolName, taints)
+
+	targetEnv, err := getVpcClusterTargetHeader(d, meta)
+	if err != nil {
+		return err
+	}
+	ClusterClient, err := meta.(conns.ClientSession).VpcContainerAPI()
```

Comment:
Is it safe to assume VpcContainerAPI here? 

It looks like the classic resource will also use this function, or am I missing something?

Reply:
if I get it right, it is not necessarily VPC. the VpcContainerAPI contains v2 api calls and most of them vpc related.
the UpdateWorkerPoolTaints() is not a vpc only one:
https://containers.cloud.ibm.com/global/swagger-global-api/#/v2/v2SetWorkerPoolTaints